src/c/src.c

```diff
@@ -162,6 +168,76 @@ void link_pass(
   /* PRINTLNC("Ending link pass"); */
 }
 
+void link_pass_ram(
+  size_t sec_len,
+  struct SectionSrc *sec_src,
+  struct SectionDst *sec_dst)
+{
+    link_pass(sec_len, sec_src, sec_dst, 0, NULL, NULL, true);
```

Comment:
```suggestion
#if defined(ST31)
  SYMBOL_ABSOLUTE_VALUE(nvram_ptr, _nvram);
  SYMBOL_ABSOLUTE_VALUE(envram_ptr, _envram);
#elif defined(ST33) || defined(ST33K1M5)
  __asm volatile("ldr %0, =_nvram":"=r"(nvram_ptr));
  __asm volatile("ldr %0, =_envram":"=r"(envram_ptr));
#else
#error "invalid architecture"
#endif

  // Value of _nvram in this run
  void* nvram_current = pic(nvram_ptr);

  link_pass(sec_len, sec_src, sec_dst, pic(nvram_ptr) - nvram_ptr, nvram_ptr, envram_ptr true);
```

Reply:
And then good to deduplicate defining `nvram_ptr` and `envram_ptr` with the other function somehow.